app/actions/requests/BasicAuthenticatedRequest.scala

```diff
@@ -36,11 +36,25 @@ case class BasicAuthenticatedRequestWithAffinityGroup[A](
       affinityGroup: AffinityGroup,
       request: Request[A]
 ) extends AuthenticatedRequest[A](request)
+
 case class BasicAuthenticatedRequest[A](
       organisationAccount: GroupAccount,
       individualAccount: DetailedIndividualAccount,
       request: Request[A]
-) extends AuthenticatedRequest[A](request)
+) extends AuthenticatedRequest[A](request) with CcaWrappedRequest {
+  override def optAccounts: Option[Accounts] =
+    Some(Accounts(organisationAccount, individualAccount))
+
+  override def isLoggedIn: Boolean = optAccounts.nonEmpty
+
+  override def yourDetailsName: Option[String] = optAccounts.map { acc =>
+    if (s"${acc.person.details.firstName} ${acc.person.details.lastName}" == acc.organisation.companyName) {
```

Comment:
this `${acc.person.details.firstName} ${acc.person.details.lastName}` appears here three times.
would be easier to see what's going on if you assigned it to a val
`val fullName = s"${acc.person.details.firstName} ${acc.person.details.lastName}"` and then used `fullName` in the three places